src/TorchSharp/DisposeScopeManager.cs

```diff
@@ -18,7 +18,7 @@ public class DisposeScopeManager
         internal ThreadDisposeScopeStatistics StatisticsInstance { get; } = new ThreadDisposeScopeStatistics();
 
         internal static DisposeScopeManager ThreadSingleton => (_threadSingleton ??= new DisposeScopeManager());
-        internal Stack<DisposeScope> DisposeScopeStack { get; } = new();
+        internal List<DisposeScope> DisposeScopeStack { get; } = new();
```

Comment:
Why is it better to use a list rather than a stack. Dispose scopes should always be LIFO.

Reply:
@NiklasGustafsson I have first created an issue for this. #1303